x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -486,41 +462,73 @@ const getDataStreams = async (
     dataStreamName: dataStream.name,
     replicated: dataStream.replicated,
     indexTemplate,
+    composedOfTemplates,
   }));
 };
 
+const getTemplateWithNameEndingIn = (composedOfTemplates: TemplateMap, suffix: string) => {
+  const mappingComponentTemplateName =
+    Object.keys(composedOfTemplates).find((templateName) => templateName.endsWith(suffix)) || '';
+
+  return composedOfTemplates[mappingComponentTemplateName];
+};
+
+const rolloverDataStream = (dataStreamName: string, esClient: ElasticsearchClient) => {
+  try {
+    // Do no wrap rollovers in retryTransientEsErrors since it is not idempotent
+    const path = `/${dataStreamName}/_rollover`;
+    return esClient.transport.request({
+      method: 'POST',
+      path,
+    });
```

Comment:
While we're here, let's go ahead and switch this to the proper rollover JS API call:

```suggestion
    return esClient.indices.rollover({
      alias: dataStreamName
    });
```